cpp/util/etcd.cc

```diff
@@ -918,5 +966,17 @@ void EtcdClient::Generic(const string& key, const string& key_space,
                       bind(&EtcdClient::FetchDone, this, etcd_req, _1)));
 }
 
+list<EtcdClient::HostPortPair> SplitHosts(const string& hosts_string) {
+  vector<string> hosts(util::split(hosts_string, ','));
+  CHECK(!hosts.empty());
+
+  list<EtcdClient::HostPortPair> ret;
+  for (const auto& h : hosts) {
+    vector<string> hp(util::split(h, ':'));
+    CHECK_EQ(2, hp.size());
+    ret.emplace_back(EtcdClient::HostPortPair(hp[0], stoi(hp[1])));
```

Comment:
With `stoi`, you might get 0 for something that doesn't parse, right? Might be a good idea to `CHECK_NE(0, port)` (with explanation)...